### ISSUE TITLE
xonfig colors and xonfig styles die in osx

### ISSUE THREAD

**Issue by santagada** [xonsh/xonsh#1412]
```bash
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 15175, in color_style_names
    return color_style_names()
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 6356, in color_style_names
    return builtins.__xonsh_shell__.shell.color_style_names()
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 15175, in color_style_names
    return color_style_names()
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 6356, in color_style_names
    return builtins.__xonsh_shell__.shell.color_style_names()
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 15175, in color_style_names
    return color_style_names()
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 6356, in color_style_names
    return builtins.__xonsh_shell__.shell.color_style_names()
RecursionError: maximum recursion depth exceeded
 ~  $  xonfig colors
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 10323, in wrapped_simple_command
    r = f(args, i)
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 14596, in xonfig
    return xonfig_main(args)
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 14137, in xonfig_main
    return _XONFIG_MAIN_ACTIONS[ns.action](ns)
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 14089, in _colors
    cmap = color_style()
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 6361, in color_style
    return builtins.__xonsh_shell__.shell.color_style()
  File "/usr/local/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 15180, in color_style
    return color_style(style=style)
TypeError: color_style() got an unexpected keyword argument 'style'
 ~  $  xonfig
+------------------+----------+
| xonsh            | 0.4.3    |
| Python           | 3.5.2    |
| PLY              | 3.7      |
| have readline    | True     |
| prompt toolkit   | 1.0.3    |
| shell type       | readline |
| pygments         | 2.1.3    |
| on posix         | True     |
| on linux         | False    |
| on darwin        | True     |
| on windows       | False    |
| on cygwin        | False    |
| is superuser     | False    |
| default encoding | utf-8    |
| git SHA          | None     |
+------------------+----------+
```

**Comment by mitnk** [comment]
Should be an amalgam issue, two `color_style()` defined (from two sources):

```
./__amalgam__.py:1085:def color_style(style='default'):
./__amalgam__.py:6412:def color_style():
```

**Comment by mitnk** [comment]
cc @scopatz

**Comment by scopatz** [comment]
@santagada @mitnk would you please try out #1415 to see if this addresses it.

**Comment by santagada** [comment]
did a python3 setup.py develop on master and now I got a traceback on start always:

```
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "/Users/santagada/devel/xonsh/xonsh/__amalgam__.py", line 10377, in wrapped_simple_command
    r = f(args, i)
  File "/Users/santagada/devel/xonsh/xonsh/__amalgam__.py", line 10775, in xontribs_main
    if not args or (args[0] not in _MAIN_ACTIONS and
NameError: name '_MAIN_ACTIONS' is not defined
```

**Comment by scopatz** [comment]
@santagada sorry that was an amalgamation failure on my part.  Fix is in #1416.

**Comment by santagada** [comment]
master fixes it for me
